src/promptflow/promptflow/_core/tools_manager.py

```diff
@@ -249,6 +251,54 @@ def _load_custom_tool(tool: Tool, node_name: str) -> Callable:
         return f_globals[func_name]
 
 
+class ToolLoader:
+    def __init__(self, working_dir: str, package_tool_keys: Optional[List[str]] = None) -> None:
+        self._working_dir = working_dir
+        self._package_tools = collect_package_tools(package_tool_keys) if package_tool_keys else {}
+
+    def load_tool_for_node(self, node: Node) -> Tool:
+        if node.source is None:
+            raise UserErrorException(f"Node {node.name} does not have source defined.")
+        if node.type is ToolType.PYTHON:
+            if node.source.type == ToolSourceType.Package:
+                return self.load_tool_for_package_node(node)
+            elif node.source.type == ToolSourceType.Code:
+                _, tool = self.load_tool_for_script_node(node)
```

Comment:
`load_tool_for_package_node` and `load_tool_for_script_node` returns different data types (one returns tool only and the other return both callable and tool). Do we have a plan to make them return the same data type?

Reply:
I am planning to refactor the code for generating tool meta, loading and resolving tools. I will try to fix this issue by the way.